server/src/main/java/org/elasticsearch/repositories/blobstore/BlobStoreRepository.java

```diff
@@ -1601,7 +1601,7 @@ public void clusterStateProcessed(String source, ClusterState oldState, ClusterS
                             } catch (IOException e) {
                                 logger.warn(() -> new ParameterizedMessage("Failed to clean up old index blobs {}", oldIndexN), e);
                             }
-                            return writtenRepositoryData;
+                            return filteredRepositoryData.withGenId(newGen);
```

Comment:
I think it's a bit weird to set this only at the very end. I think we shuld set the new genId on the RepositoryData object as soon as we have `final long newGen = setPendingStep.result();`?

Reply:
Fair point, I pushed 602e32f413a to clean this up :)